src/app/settings/views/Configuration/Security/SessionTimeout/SessionTimeout.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+
+import { Spinner, Select, Button, Icon } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as configActions } from "app/store/config";
+import configSelectors from "app/store/config/selectors";
+
+type SessionTimeoutFormValues = {
+  session_length: number;
+  time_unit: "hours" | "days";
+};
+
+export enum Labels {
+  Loading = "Loading...",
+  Edit = "Edit",
+  Expiration = "Session timeout expiration",
+  Save = "Save",
+  Cancel = "Cancel",
+  TimeUnit = "Time unit",
+  Days = "Day(s)",
+  Hours = "Hour(s)",
+  ConfigureSessionTimeout = "Configure Session Timeout",
+}
+
+const SessionTimeout = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const configLoading = useSelector(configSelectors.loading);
+  const sessionLength = useSelector(configSelectors.sessionLength) || 1209600; // fallback to default value of 14 days if none is set
```

Comment:
The default fallback should only be used for the edit form as a default if none has been set - as I understand we're setting a default suggested value.

This will display a value in a non-editable view which makes user think that that's currently saved setting for this option and is misleading.

Reply:
In any case, please use a util function such as `daysToSeconds` instead to make this clearer. Feel free to create it if doesn't exist, we have a few custom utils in the `timeSpan.ts` file.

Reply:
Oh right - since sessionLength is possibly null, what should I do in that case?

Reply:
Would `const sessionLength = useSelector(configSelectors.sessionLength) || undefined;` be acceptable?